lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -356,7 +356,9 @@
     required: yes
   networks:
     description:
-      - List of the service networks names.
+      - List of the service networks names or dictionaries.
+      - When passed dictionaries valid sub-options are C(name) which is required and
+        C(aliases) and C(options).
       - Prior to API version 1.29, updating and removing networks is not supported.
         If changes are made the service will then be removed and recreated.
```

Comment:
BTW, when I read this part I just thought about a new feature, which would also be useful for `docker_container`: `fail_instead_of_restart` (boolean, default `false`). If the module thinks it needs to recreate the service/container/..., fail the module instead of doing it. This setting can be used for important containers; then users will know when they accidentally changed something important and won't just kill the service/container.

No idea if that's really useful, just wanted to write it down since it just came to my mind :)

Reply:
Yes that might be a good option to add. Services will be recreated then `mode` and `networks` are changed (only < 1.29).